patches/chrome-installer-mini_installer-mini_installer.cc.patch

```diff
@@ -11,23 +19,24 @@ index 73e6240800f3c72bd96c91971c53a19516e1b0f1..e0049fab53dcb2bb8fe237523bf6eb98
  // Opens the Google Update ClientState key. If |binaries| is false, opens the
  // key for Google Chrome or Chrome SxS (canary). If |binaries| is true and an
  // existing multi-install Chrome is being updated, opens the key for the
-@@ -144,6 +144,138 @@ void SetInstallerFlags(const Configuration& configuration) {
+@@ -144,6 +145,141 @@ void SetInstallerFlags(const Configuration& configuration) {
  }
  #endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING)
  
 +#if defined(BRAVE_CHROMIUM_BUILD)
 +bool ParseStandardReferralCode(const wchar_t* filename, ReferralCodeString& referral_code) {
 +  // Scan backwards for last dash in filename.
-+  const wchar_t* scan = filename + lstrlen(filename) - 1;
++  const wchar_t* scan = filename + lstrlenW(filename) - 1;
 +  while (scan != filename && *scan != L'-')
 +    --scan;
 +
 +  if (*scan++ != L'-')
 +    return false;
 +
 +  // Standard referral code is 6 characters.
-+  const wchar_t* ref_code = scan;
-+  if (lstrlen(ref_code) != 6)
++  wchar_t* ref_code = scan;
++  wchar_t ref_code_normalized[StandardReferralCodeLen];
++  if (lstrlenW(ref_code) != StandardReferralCodeLen)
```

Comment:
Should probably call `SafeStrLen` here and elsewhere in this file; there are some rules about not pulling extraneous run-time code into the mini-installer (see the comment header in `mini_installer.cc`).

Reply:
Excellent. I see there's also a `SafeStrCpy` that I can use instead of `lstrcpyW`. I don't see any equivalent for `CharUpperW` unfortunately.

Reply:
We could roll our own, I don't think the referral code is expected to be anything other than ASCII characters.